x-pack/test/apm_api_integration/tests/alerts/transaction_error_rate.spec.ts

```diff
@@ -238,82 +176,49 @@ export default function ApiTest({ getService }: FtrProviderContext) {
         });
         expect(createdRule.id).to.not.eql(undefined);
-        ruleId2 = createdRule.id;
+        ruleId = createdRule.id;
       });
 
-      it('checks if rule is active', async () => {
-        const executionStatus = await waitForRuleStatus({
-          id: ruleId2,
-          expectedStatus: 'active',
-          supertest,
-        });
-        expect(executionStatus.status).to.be('active');
+      after(async () => {
+        await deleteRuleById({ supertest, ruleId });
+        await deleteAlertsByRuleId({ es, ruleId });
       });
 
       it('indexes alert document with all group-by fields', async () => {
-        const resp = await waitForAlertInIndex({
-          es,
-          indexName: APM_ALERTS_INDEX,
-          ruleId: ruleId2,
-        });
-        alertId = (resp.hits.hits[0]._source as any)['kibana.alert.uuid'];
-        startedAt = (resp.hits.hits[0]._source as any)['kibana.alert.start'];
-
-        expect(resp.hits.hits[0]._source).property('service.name', 'opbeans-node');
-        expect(resp.hits.hits[0]._source).property('service.environment', 'production');
-        expect(resp.hits.hits[0]._source).property('transaction.type', 'request');
-        expect(resp.hits.hits[0]._source).property('transaction.name', 'tx-node');
+        const alerts = await waitForAlertsForRule({ es, ruleId });
+        expect(alerts[0]).property('service.name', 'opbeans-node');
+        expect(alerts[0]).property('service.environment', 'production');
+        expect(alerts[0]).property('transaction.type', 'request');
+        expect(alerts[0]).property('transaction.name', 'tx-node');
       });
 
-      it('returns correct message', async () => {
-        const rangeFrom = moment(startedAt).subtract('5', 'minute').toISOString();
-        const resp = await waitForDocumentInIndex<{ message: string }>({
-          es,
-          indexName: ALERT_ACTION_INDEX_NAME2,
-        });
-
-        expect(resp.hits.hits[0]._source?.message).eql(`Transaction Name: tx-node
-- Alert URL: http://mockedpublicbaseurl/app/observability/alerts?_a=(kuery:%27kibana.alert.uuid:%20%22${alertId}%22%27%2CrangeFrom:%27${rangeFrom}%27%2CrangeTo:now%2Cstatus:all)`);
+      it('produces an alert with the correct reason', async () => {
+        const alerts = await waitForAlertsForRule({ es, ruleId });
+        expect(alerts[0]!['kibana.alert.reason']).to.be(
+          'Failed transactions is 50% in the last 5 mins for service: opbeans-node, env: production, type: request, name: tx-node. Alert when > 40%.'
+        );
       });
 
-      it('shows the correct alert count for each service on service inventory', async () => {
+      it('shows alert count=1 for opbeans-node on service inventory', async () => {
         const serviceInventoryAlertCounts = await fetchServiceInventoryAlertCounts(apmApiClient);
         expect(serviceInventoryAlertCounts).to.eql({
           'opbeans-node': 1,
-          'opbeans-java': 1,
+          'opbeans-java': 0,
```

Comment:
This was an incorrect assertion caused by the alert pollution. Now alerts are cleaned up between runs.